### ISSUE TITLE
make geth fails Ubuntu 16.04 64bit go 1.9

### ISSUE THREAD

**Issue by IKqlXKPyQHlHrv1YI71Hw829W9uaLOjE** [ethereum/go-ethereum#15752]
```
root@h2752422:~/go-ethereum# make geth
build/env.sh go run build/ci.go install ./cmd/geth
>>> /usr/local/go/bin/go install -ldflags -X main.gitCommit=9d187f02389ba12493112c7feb15a83f44e3a3ff -v ./cmd/geth
github.com/ethereum/go-ethereum/vendor/gopkg.in/olebedev/go-duktape.v3
# github.com/ethereum/go-ethereum/vendor/gopkg.in/olebedev/go-duktape.v3
In file included from vendor/gopkg.in/olebedev/go-duktape.v3/api.go:7:0:
vendor/gopkg.in/olebedev/go-duktape.v3/api.go: In function ‘_duk_error’:
vendor/gopkg.in/olebedev/go-duktape.v3/duktape.h:510:127: warning: right-hand operand of comma expression has no effect [-Wunused-value]
  (duk_error_raw((ctx), (duk_errcode_t) (err_code), (const char *) (DUK_FILE_MACRO), (duk_int_t) (DUK_LINE_MACRO), __VA_ARGS__), (duk_ret_t) 0)
                                                                                                                               ^
vendor/gopkg.in/olebedev/go-duktape.v3/api.go:153:2: note: in expansion of macro ‘duk_error’
  duk_error(ctx, err_code, "%s", str);
  ^
# github.com/ethereum/go-ethereum/vendor/gopkg.in/olebedev/go-duktape.v3
vendor/gopkg.in/olebedev/go-duktape.v3/duktape.go:132: constant 18446744073709551615 overflows int64
util.go:45: exit status 2
exit status 1
Makefile:15: recipe for target 'geth' failed
make: *** [geth] Error 1
root@h2752422:~/go-ethereum# go version
go version go1.9.1 linux/amd64
root@h2752422:~/go-ethereum# go env
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/root/go"
GORACE=""
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build854197436=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
root@h2752422:~/go-ethereum#

```

**Comment by leo-xinwang** [comment]
The same happens on macOS 10.13.2

**Comment by leo-xinwang** [comment]
I find a work around to solve this:
change line 192 of vendor/gopkg.in/olebedev/go-duktape.v3
from:
func (d *Context) PushCFunction(fn *[0]byte, nargs int64) int {
to:
func (d *Context) PushCFunction(fn *[0]byte, nargs uint64) int {

The reason seems go has problem in interpret an negative constant integer in C.

**Comment by leo-xinwang** [comment]
The problem is gone under go 1.9.2

**Comment by dherbst** [comment]
Please consider closing this issue, upgrading to go1.9.2 fixes this problem.

**Comment by Ajaysethi136** [comment]
build/env.sh go run build/ci.go install ./cmd/geth
ln: failed to create symbolic link 'go-ethereum/.': File exists
make: *** [Makefile:15: geth] Error in Windows 10
Could someone plz help

**Comment by danclegg** [comment]
I can also confirm the fix from @leo-xinwang -- upgrading to 1.9.2 fixes this issue.

**Comment by Ajaysethi136** [comment]
I'm using go1.9.2 version only. Plz help

**Comment by danclegg** [comment]
@Ajaysethi136 It seems like you just need to remove the existing symlink. Running this in the install location should work:
`rm go-ethereum`

Then run the build command again.

**Comment by Ajaysethi136** [comment]
I tried removing but it throws the following error "rm: cannot remove 'go-ethereum': No such file or directory". 
Currently, I'm using Windows 10, Might it causing be the issue?? or should we use Ubuntu only for this?

**Comment by jsp2012** [comment]
@leo-xinwang I`m sorry I did not know which file you mean.In my env go-duktape.v3 is a folder. and i find func (d *Context) PushCFunction(fn *[0]byte, nargs int64) int {  in the 976 lines of vendor/gopkg.in/olebedev/go-duktape.v3/api.go  and I change it to uint but still get the error vendor/gopkg.in/olebedev/go-duktape.v3/api.go:153:2: note: in expansion of macro ‘duk_error’
  duk_error(ctx, err_code, "%s", str);

**Comment by ronaldpetty** [comment]
I am using ubuntu 16.04.1, go 1.10, geth 1.8.1 . I also see the same error, I have tried to modify to unit and it fails as follows:

```
...
vendor/gopkg.in/olebedev/go-duktape.v3/duktape.go:132: constant -1 overflows uint64
```

**Comment by tech-alchemist** [comment]
v1.9.2 worked for CentOS too.

**Comment by jollysean** [comment]
The issue exists in go version go1.13.1 linux/amd64

**Comment by jyap808** [comment]
Fix integrated in Geth with #20777
